### ISSUE TITLE
Getting segmentation fault in using sr_xpath_node_name API

### ISSUE THREAD

**Issue by rinkalbhatia06** [sysrepo/sysrepo#2133]
char *xp_node_name = NULL;

    xp_node_name = sr_xpath_node_name(value->xpath);

    if (NULL == xp_node_name)
    {

    }
    else
    {
        printf("%s@@@\n", xp_node_name); //segmentation fault coming in this line
    }

**Comment by michalvasko** [comment]
What *sysrepo* version are you using? What is the input (`value->xpath`) it crashes with?

**Comment by rinkalbhatia06** [comment]
sysrepo version = 1.4.70
value->xpath = /wlm-rx-power:wlmdetails/wlmdetail[rackId='6'][subRackId='1'][cardId='2'][cardType='53'][cardSubType='0'][portId='1']/rackId
wlm-rx-power.yang

**Comment by rinkalbhatia06** [comment]
module wlm-rx-power {
    namespace "urn:sysrepo:wlm-rx-power";
    prefix wlmrxpow;

    container wlmdetails{
    	list wlmdetail{
    		key "rackId subRackId cardId cardType cardSubType portId";
	    	leaf rackId{
	    		type uint8;
	    	}

	    	leaf subRackId{
	    		type uint8;
	    	}

	    	leaf cardId{
	    		type uint8;
	    	}

	    	leaf cardType{
	    		type uint8;
	    	}

	    	leaf cardSubType{
	    		type uint8;
	    	}

	    	leaf portId{
	    		type uint8;
	    	}

	    	leaf waveCode{
	    		type uint8;
	    	}

	    	leaf direction{
	    		type uint8;
	    	}

	    	leaf rxPower{
	         type decimal64{
	         	fraction-digits 2;
	         }
	    	}

	    	leaf losState{
	    		type uint8;
	    	}
    	}
    }
}

**Comment by michalvasko** [comment]
Hi,
I have tested it and it works fine for me. Could you try to run your code with `valgrind`, for example? I think the problem is happening before, it just manifests later.

**Comment by rinkalbhatia06** [comment]
Hi Michal,
Thanks for quick response,
![image](https://user-images.githubusercontent.com/69560210/92585323-72106900-f2b2-11ea-9610-9489984b46c1.png)

This error is showing using gdb

**Comment by michalvasko** [comment]
Hi,
that is not helpful, only `valgrind` (or some other similar tool such as ASan) output could reveal the problem directly.

**Comment by rinkalbhatia06** [comment]
![image](https://user-images.githubusercontent.com/69560210/92591996-7f325580-f2bc-11ea-9ebb-b0609d891d66.png)

**Comment by michalvasko** [comment]
Well, this output says you are accessing a `NULL` pointer and I think that is a problem in your code.

**Comment by rinkalbhatia06** [comment]
Thanks Michal, problem solved

**Comment by rinkalbhatia06** [comment]
Hi Michal,

sudo ./application_changes_example wlm-rx-power

Output:
![image](https://user-images.githubusercontent.com/69560210/92681440-3fff1580-f34b-11ea-8493-67859eb8d7be.png)

sudo valgrind --tool=memcheck ./application_changes_example wlm-rx-power

Output:
![image](https://user-images.githubusercontent.com/69560210/92681488-5efda780-f34b-11ea-9e5c-bf35cd01a826.png)

application is running fine with valgrind

Code: 
void fill_val_in_wlm_details_str(const sr_val_t *value)
{
    if (NULL == value) 
    {
        return;
    }

    char *xp_node_name = NULL;

    xp_node_name = sr_xpath_node_name(value->xpath);
    
    if (NULL != xp_node_name)
    {
        if(strcmp(xp_node_name, "rackId") == 0)
        {
            wlmDetails.rackId = value->data.int8_val;
        }
        else if(strcmp(xp_node_name, "subRackId") == 0)
        {
            wlmDetails.subRackId = value->data.int8_val;
        }
        else if(strcmp(xp_node_name, "cardId") == 0)
        {
            wlmDetails.cardId = value->data.int8_val;
        }
        else if(strcmp(xp_node_name, "cardType") == 0)
        {
            wlmDetails.cardType = value->data.int8_val;
        }
        else if(strcmp(xp_node_name, "cardSubType") == 0)
        {
            wlmDetails.cardSubType = value->data.int8_val;
        }
        else if(strcmp(xp_node_name, "portId") == 0)
        {
            wlmDetails.portId = value->data.int8_val;
        }
        else if(strcmp(xp_node_name, "waveCode") == 0)
        {
            wlmDetails.waveCode = value->data.int8_val;
        }
        else if(strcmp(xp_node_name, "direction") == 0)
        {
            wlmDetails.direction = value->data.int8_val;
        }
        else if(strcmp(xp_node_name, "rxPower") == 0)
        {
            wlmDetails.power = value->data.decimal64_val;
        }
        else if(strcmp(xp_node_name, "losState") == 0)
        {
            wlmDetails.losState = value->data.int8_val;
        }
    }
}

**Comment by michalvasko** [comment]
In case it is running fine with `valgrind` and crashes without it, then I have no way of helping you. You will have to find the problem yourself, but in case it crashes also with `gdb`, it should be trivial to find where the `NULL` pointer is being accessed.

**Comment by rinkalbhatia06** [comment]
Hi Michal,

int
main(int argc, char **argv)
{
    /*#############################################################*/
    char *xp_node_name = NULL;
    char xpath1[400];
    memset(xpath1,0,400);
    strcpy(xpath1,"/wlm-rx-power:wlmdetails/wlmdetail[rackId='6'][subRackId='1'][cardId='2'][cardType='53'][cardSubType='0'][portId='1']/cardSubType");
    xp_node_name = sr_xpath_node_name(xpath1);
    printf("%s\n", xp_node_name); //Line number=233
    exit(0);
    /*#############################################################*/
}

I just wrote this code now. But segmentation fault is still coming. 

This is gdb output:
![image](https://user-images.githubusercontent.com/69560210/92861261-e9f7a400-f416-11ea-81be-815392e3c2bb.png)

**Comment by michalvasko** [comment]
Okay, I think I have found the problem and you have most likely just wasted my time. Can you paste the output of this example (or even your original app) compilation?

**Comment by rinkalbhatia06** [comment]
make output:
![image](https://user-images.githubusercontent.com/69560210/92879005-42846c80-f42a-11ea-9ea9-48cd71b7f199.png)

**Comment by michalvasko** [comment]
Yes, well, if you are not bothered by compilation warnings, I cannot help you.
